Software/src/main/java/org/foi/emp/carmanagement/configs/JwtRequestFilter.java

```diff
@@ -0,0 +1,62 @@
+package org.foi.emp.carmanagement.configs;
+
+import io.jsonwebtoken.ExpiredJwtException;
+import org.foi.emp.carmanagement.services.JwtUserDetailsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+public class JwtRequestFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private JwtUserDetailsService jwtUserDetailsService;
+
+    @Autowired
+    private JwtTokenUtil jwtTokenUtil;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain) throws IOException, ServletException
+
+    {
+        final String requestTokenHeader = response.getHeader("Authorization");
```

Comment:
Make a class constant at the top of the class.
`private static final String AUTHORIZATION = "Authorization";`
and use it here as 
`final String requestTokenHeader = response.getHeader(AUTHORIZATION);`